--- a/fn/-z4h-comp-files
+++ b/fn/-z4h-comp-files
@@ -7,13 +7,13 @@

   if (( _z4h_only_dirs )); then
     if (( dot_glob )); then
-      local dirs=($_z4h_path_prefix${^${(Q)_z4h_words:#.*}}/*(D-/Y1N:h:t))
+      local dirs=($_z4h_path_prefix${^${(Q)_z4h_words:#.git}}/*(D-/Y1N:h:t))
     else
       local dirs=($_z4h_path_prefix${^${(Q)_z4h_words:#.*}}/*(-/Y1N:h:t))
     fi
   else
     if (( dot_glob )); then
-      local dirs=($_z4h_path_prefix${^${(Q)_z4h_words:#.*}}/*(DY1N:h:t))
+      local dirs=($_z4h_path_prefix${^${(Q)_z4h_words:#.git}}/*(DY1N:h:t))
     else
       local dirs=($_z4h_path_prefix${^${(Q)_z4h_words:#.*}}/*(Y1N:h:t))
     fi
@@ -42,7 +42,7 @@
       } | {
         -z4h-cursor-show
         2>$TTY -z4h-fzf                                     \
-          --query=${_z4h_word_prefix:+"^$_z4h_word_prefix"} \
+          --query="$_z4h_word_prefix"                       \
           --color=hl:201,hl+:201                            \
           --with-nth=2                                      \
           --delimiter='\000'                                \
--- a/fn/-z4h-comp-words
+++ b/fn/-z4h-comp-words
@@ -52,7 +52,7 @@
       } | {
         -z4h-cursor-show
         2>$TTY -z4h-fzf                                     \
-          --query=${_z4h_word_prefix:+"^$_z4h_word_prefix"} \
+          --query="$_z4h_word_prefix"                       \
           --color=hl:201,hl+:201                            \
           --with-nth=2                                      \
           --delimiter='\000'                                \
--- a/fn/z4h-cd-down
+++ b/fn/z4h-cd-down
@@ -19,7 +19,7 @@ local -i list_colors=$((!$?))
       eval "$_z4h_opt"
       if (( dot_glob )); then
         local dirs=(./*(-/DN))
-        local non_empty=(${^${dirs:#./.*}}/*(D-/Y1N:h:t))
+        local non_empty=(${^${dirs:#./.git}}/*(D-/Y1N:h:t))
       else
         local dirs=(./*(-/N))
         local non_empty=(${^dirs}/*(-/Y1N:h:t))
